### PR TITLE
Comment NL27

### PR DESCRIPTION
**Pull request by Aurige**
Addition of TypeOfProductCategoryRef  and TypeOfServiceRef on a Line (requires to move them from Part 2 to Part 1)

### CODE REVIEW

**Comment by skinkie** [comment]
Lacks documentation only available;
image

**Comment by skinkie** [comment]
Entire XSD update has been added by @nick-knowles, documentation is not available.

File: xsd/netex_part_1/part1_networkDescription/netex_line_support.xsd [hunk 1]
```diff
@@ -381,4 +381,81 @@ Rail transport, Roads and Road transport
 			</xsd:restriction>
 		</xsd:simpleContent>
 	</xsd:complexType>
+	<!-- ======================================================================= -->
+	<xsd:simpleType name="TypeOfServiceIdType">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_support.xsd:385]: The first 4 elements are all defined in netex_serviceRestrictions_support.xsd => don't do those here.

File: xsd/netex_part_1/part1_networkDescription/netex_line_support.xsd [hunk 2]
```diff
@@ -381,4 +381,81 @@ Rail transport, Roads and Road transport
...
+		<xsd:complexContent>
+			<xsd:extension base="oneToManyRelationshipStructure">
+				<xsd:sequence>
+					<xsd:element ref="TypeOfServiceRef" minOccurs="0" maxOccurs="unbounded"/>
+				</xsd:sequence>
+			</xsd:extension>
+		</xsd:complexContent>
+	</xsd:complexType>
+	<!-- ======================================================================= -->
+	<xsd:simpleType name="TypeOfProductCategoryIdType">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_support.xsd:423]: Those are also all in netex_serviceRestrictions_support.xsd. First file already integrated in master.

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 3]
```diff
@@ -181,6 +186,18 @@ Rail transport, Roads and Road transport
 			</xsd:extension>
 		</xsd:complexContent>
 	</xsd:complexType>
+	<xsd:complexType name="typesOfServiceInFrame_RelStructure">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:189]: in netex_serviceRestrictions_version

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 4]
```diff
@@ -131,6 +131,11 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>DESTINATION DISPLAYs in frame.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element name="typesOfServices" type="typesOfServiceInFrame_RelStructure" minOccurs="0">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:134]: This part is not found.... If needed it stands. @Aurige

Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:134]: This part is not found.... If needed it stands. @Aurige

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 5]
```diff
@@ -428,6 +445,8 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>Additional transport MODEs for LINE.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element ref="TypeOfProductCategoryRef" minOccurs="0"/>
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:448]: This part is also not in master currently.

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 6]
```diff
@@ -1130,4 +1149,89 @@ Rail transport, Roads and Road transport
 		</xsd:complexContent>
 	</xsd:complexType>
 	<!-- ======================================================================= -->
+	<xsd:element name="TypeOfProductCategory" abstract="false" substitutionGroup="TypeOfValue">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1152]: TypeOfProductCategory already in master in netex_serviceRestrictions_version.xd

Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1152]: TypeOfProductCategory already in master in netex_serviceRestrictions_version.xd

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 7]
```diff
@@ -1130,4 +1149,89 @@ Rail transport, Roads and Road transport
...
+	<!-- ======================================================================= -->
+	<xsd:element name="TypeOfService" abstract="false" substitutionGroup="TypeOfEntity">
+		<xsd:annotation>
+			<xsd:documentation>Classification of a Service.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:complexType>
+			<xsd:complexContent>
+				<xsd:restriction base="TypeOfServiceStructure">
+					<xsd:sequence>
+						<xsd:sequence>
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1212]: TypeOfService is in master in netex_serviceRestrictions_version and even with more content:
At this place the version here is missing:
						<xsd:sequence>
							<xsd:group ref="EntityInVersionGroup" minOccurs="0"/>
						</xsd:sequence>

Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1212]: TypeOfService is in master in netex_serviceRestrictions_version and even with more content:
At this place the version here is missing:
						<xsd:sequence>
							<xsd:group ref="EntityInVersionGroup" minOccurs="0"/>
						</xsd:sequence>

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 8]
```diff
@@ -1130,4 +1149,89 @@ Rail transport, Roads and Road transport
...
+					<xsd:attribute name="id" type="TypeOfServiceIdType">
+						<xsd:annotation>
+							<xsd:documentation>Reference to a TYPE OF SERVICE.</xsd:documentation>
+						</xsd:annotation>
+					</xsd:attribute>
+				</xsd:restriction>
+			</xsd:complexContent>
+		</xsd:complexType>
+	</xsd:element>
+	<xsd:complexType name="TypeOfServiceStructure" abstract="false">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1228]: exists already in master TypeOfServiceStructure

Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:1228]: exists already in master TypeOfServiceStructure

File: xsd/netex_part_2/part2_journeyTimes/netex_journey_version.xsd [hunk 9]
```diff
@@ -8,6 +8,7 @@
 	<xsd:include schemaLocation="../../netex_framework/netex_reusableComponents/netex_noticeAssignment_version.xsd"/>
 	<xsd:include schemaLocation="../../netex_framework/netex_reusableComponents/netex_mode_version.xsd"/>
 	<xsd:include schemaLocation="netex_journeyAccounting_version.xsd"/>
+	<xsd:include schemaLocation="../../netex_part_1/part1_networkDescription/netex_line_support.xsd"/>
```
Comment (ue71603) [inline xsd/netex_part_2/part2_journeyTimes/netex_journey_version.xsd:11]: Didn't find it in master. But it seed to have worked just fine. Even also without netex_serviceRestrictions_support.xsd. At least this needs to be rechecked, if it should stay in the pull request.

Comment (ue71603) [inline xsd/netex_part_2/part2_journeyTimes/netex_journey_version.xsd:11]: Didn't find it in master. But it seed to have worked just fine. Even also without netex_serviceRestrictions_support.xsd. At least this needs to be rechecked, if it should stay in the pull request.

**Review by ue71603 — CHANGES_REQUESTED** [review]
The PR is mostly done already.  The rest needs to be rechecked and resubmitted.

**Comment by ue71603** [comment]
This is one of the old CR.
•	I think first Christophe needs to check, which of the things we should still add
•	Then André can check if everything is in the documentation (from the things we don’t want to add, because they should be there)
•	Then we shrink the CR to what still needs to be done
•	Then we merge it
•	Then we update the rest of the documentation.

**Comment by Aurige** [comment]
It is already in next so we can close it

**Comment by trurlurl** [comment]
Removing "needs documentation update" label since closed without having been merged.

**Pull request closed without merging**